adapter/src/test/kotlin/org/javacs/ktda/GlobalWorkspaceTest.kt

```diff
@@ -0,0 +1,242 @@
+        )
+    var client = mockk<IDebugProtocolClient>(relaxed = true)
+    lateinit var debugAdapter: KotlinDebugAdapter
+    val latch = CountDownLatch(1)
+
+    @Before
+    fun startDebugAdapter() {
+        // Reset the mock debug protocol client
+        client = mockk(relaxed = true)
+
+        // Build the project first
+        val process = ProcessBuilder(buildCommand.first, buildCommand.second)
+            .directory(absoluteWorkspaceRoot.toFile())
+            .inheritIO()
+            .start()
+        process.waitFor()
+        assertThat(process.exitValue(), equalTo(0))
+
+        debugAdapter = JDILauncher()
+            .let(::KotlinDebugAdapter)
+            .also {
+                it.connect(client)
+                val configDone = it.configurationDone(ConfigurationDoneArguments())
+                it.initialize(InitializeRequestArguments().apply {
+                    adapterID = "test-debug-adapter"
+                    linesStartAt1 = true
+                    columnsStartAt1 = true
+                }).join()
+                // Slightly hacky workaround to ensure someone is
+                // waiting on the ConfigurationDoneResponse. See
+                // KotlinDebugAdapter.kt:performInitialization for
+                // details.
+                Thread {
+                    configDone.join()
+                }.start()
+                // Wait until the thread has blocked on the future
+                while (configDone.numberOfDependents == 0) {
+                    Thread.sleep(100)
+                }
+            }
+
+        // Print output on commandline for test debugging
+        slot<OutputEventArguments>().also { outputEvent ->
+            every { client.output(capture(outputEvent)) }.answers { println(outputEvent.captured.output) }
+        }
+        // And release the latch when the debug session is terminated/Done
+        every { client.terminated(any()) }.answers { latch.countDown() }
+    }
+
+    @After
+    fun closeDebugAdapter() {
+        debugAdapter.disconnect(DisconnectArguments()).join()
+    }
+
+    @Test
+    fun `Breakpoints should be hit and variables recorded`() {
+        // Given: We set a breakpoint
+        debugAdapter.setBreakpoints(SetBreakpointsArguments().apply {
+            source = Source().apply {
+                name = "App.kt"
+                path = absoluteWorkspaceRoot
+                    .resolve("src")
+                    .resolve("main")
+                    .resolve("kotlin")
+                    .resolve("sample")
+                    .resolve("workspace")
+                    .resolve("App.kt")
+                    .toString()
+            }
+            breakpoints = arrayOf(SourceBreakpoint().apply {
+                line = 8
+            })
+        }).join()
+        var locals: List<Variable> = emptyList()
+
+        // TODO: We need this workaround because for some reason when the breakpoint is hit two StoppedEvents are send
```

Comment:
I will see if I can verify if this is an actual bug or no. Just wanted to show you the stuff I have first